cvat/apps/consensus/merge_consensus_jobs.py

```diff
@@ -0,0 +1,102 @@
+from typing import List, Dict
+import datumaro as dm
+import django_rq
+from django.conf import settings
+from datumaro.components.operations import IntersectMerge
+from cvat.apps.quality_control.quality_reports import JobDataProvider
+from cvat.apps.engine.models import Job, JobType
+from cvat.apps.dataset_manager.bindings import import_dm_annotations
+from rest_framework import status
+from rest_framework.response import Response
+from cvat.apps.dataset_manager.task import patch_job_data, PatchAction
+from cvat.apps.engine.utils import get_rq_lock_by_user, get_rq_job_meta, define_dependent_job, process_failed_job
+from cvat.apps.engine.serializers import RqIdSerializer
+from django.utils import timezone
+from django.db import transaction
+
+def get_consensus_jobs(task_id: int):
+    jobs = {} # parent_job_id -> [consensus_job_id]
+    for job in Job.objects.select_related("segment").filter(segment__task_id=task_id, type=JobType.ANNOTATION.value).order_by('id'):
+        if job.parent_job_id is not None:
+            if job.parent_job_id not in jobs:
+                jobs[job.parent_job_id] = []
+            jobs[job.parent_job_id].append(job.id)
+    return jobs
+
+def get_annotations(job_id: int):
+    return JobDataProvider(job_id).dm_dataset
+
+
+@transaction.atomic
+def _merge_consensus_jobs(task_id: int):
+    jobs = get_consensus_jobs(task_id)
+    merger = IntersectMerge()
+
+    for parent_job_id, job_ids in jobs.items():
+        consensus_dataset = list(map(get_annotations, job_ids))
+
+        merged_dataset: dm.Dataset = merger(consensus_dataset)
+
+        # check if the merged dataset has annotations
+        for item in merged_dataset:
+            if not item.annotations:
+                return 400
+
+        # delete the existing annotations in the job
+        patch_job_data(parent_job_id, None, PatchAction.DELETE)
+        """
+        if we don't delete exising annotations, the imported annotations
+        will be appended to the existing annotations, and thus updated annotation
+        would have both exisiting + imported annotations, but we only want the
+        imported annotations
+        """
+
+        parent_job = JobDataProvider(parent_job_id)
+
+        # imports the annotations in the `parent_job.job_data` instance
+        import_dm_annotations(merged_dataset, parent_job.job_data)
+
+        # updates the annotations in the job
+        patch_job_data(parent_job_id, parent_job.job_data.data.serialize(), PatchAction.UPDATE)
+        return 201
```

Comment:
If you want to use HTTP status codes here, please use the named constants instead. However, I'd suggest to return None or throw exceptions on errors. The exceptions will be thrown anyway, as you're not catching them here. Unlike exceptions, return codes will make the job `finished`, instead of `failed`, so the result parsing logic is more complex.